From d917df594b2585dbb86fe273eb3e59f18e3567b9 Mon Sep 17 00:00:00 2001
From: Alexander Strange <astrange@ithinksw.com>
Date: Sun, 22 Nov 2009 04:09:09 -0500
Subject: [PATCH 2/6] Remove the warning "Cannot parallelize deblocking type
 1".

---
 libavcodec/h264.c |    5 +----
 1 files changed, 1 insertions(+), 4 deletions(-)

diff --git a/libavcodec/h264.c b/libavcodec/h264.c
index acd7179..7d27caa 100644
--- a/libavcodec/h264.c
+++ b/libavcodec/h264.c
@@ -3144,10 +3144,7 @@ static int decode_slice_header(H264Context *h, H264Context *h0){
             h->deblocking_filter = 2;
         } else {
             h0->max_contexts = 1;
-            if(!h0->single_decode_warning) {
-                av_log(s->avctx, AV_LOG_INFO, "Cannot parallelize deblocking type 1, decoding such frames in sequential order\n");
-                h0->single_decode_warning = 1;
-            }
+
             if (h != h0) {
                 av_log(h->s.avctx, AV_LOG_ERROR, "Deblocking switched inside frame.\n");
                 return 1;
-- 
1.7.8

