From 05d311486ffbd12d3afa9af6d1b4eeff98deffaa Mon Sep 17 00:00:00 2001
From: Alexander Strange <astrange@ithinksw.com>
Date: Sat, 6 Nov 2010 05:24:09 -0400
Subject: [PATCH 5/6] Workaround for AVI audio tracks importing 1152x too long

---
 libavformat/avidec.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libavformat/avidec.c b/libavformat/avidec.c
index ad35995..8ab97b0 100644
--- a/libavformat/avidec.c
+++ b/libavformat/avidec.c
@@ -98,7 +98,7 @@ static int guess_ni_flag(AVFormatContext *s);
 
 static inline int get_duration(AVIStream *ast, int len){
     if(ast->sample_size){
-        return len;
+        return len / ast->sample_size;
     }else if (ast->dshow_block_align){
         return (len + ast->dshow_block_align - 1)/ast->dshow_block_align;
     }else
-- 
1.7.8

