Index: libavformat/avidec.c
===================================================================
--- a/libavformat/avidec.c	(revision 25498)
+++ b/libavformat/avidec.c	(working copy)
@@ -97,7 +97,7 @@
 
 static inline int get_duration(AVIStream *ast, int len){
     if(ast->sample_size){
-        return len;
+        return len / ast->sample_size;
     }else if (ast->dshow_block_align){
         return (len + ast->dshow_block_align - 1)/ast->dshow_block_align;
     }else
