Index: ffmpeg/libavcodec/h264.h
===================================================================
--- ffmpeg/libavcodec/h264.h	(revision 22107)
+++ ffmpeg/libavcodec/h264.h	(working copy)
@@ -57,10 +57,13 @@
 
 /* Compiling in interlaced support reduces the speed
  * of progressive decoding by about 2%. */
-#define ALLOW_INTERLACE
+//#define ALLOW_INTERLACE
 
-#define ALLOW_NOCHROMA
+//#define ALLOW_NOCHROMA
 
+#undef CODEC_FLAG2_CHUNKS
+#define CODEC_FLAG2_CHUNKS 0
+
 #define FMO 0
 
 /**
