Index: ffmpeg/libavcodec/h264.c
===================================================================
--- ffmpeg/libavcodec/h264.c	(revision 18866)
+++ ffmpeg/libavcodec/h264.c	(working copy)
@@ -4090,10 +4090,7 @@
             h->deblocking_filter = 2;
         } else {
             h0->max_contexts = 1;
-            if(!h0->single_decode_warning) {
-                av_log(s->avctx, AV_LOG_INFO, "Cannot parallelize deblocking type 1, decoding such frames in sequential order\n");
-                h0->single_decode_warning = 1;
-            }
+
             if(h != h0)
                 return 1; // deblocking switched inside frame
         }
