Index: ffmpeg/libavcodec/mpegvideo.h
===================================================================
--- ffmpeg/libavcodec/mpegvideo.h	(revision 21580)
+++ ffmpeg/libavcodec/mpegvideo.h	(working copy)
@@ -56,7 +56,7 @@
 
 #define MAX_THREADS 16
 
-#define MAX_PICTURE_COUNT 32
+#define MAX_PICTURE_COUNT 64
 
 #define ME_MAP_SIZE 64
 #define ME_MAP_SHIFT 3
Index: ffmpeg/libavcodec/utils.c
===================================================================
--- ffmpeg/libavcodec/utils.c	(revision 21580)
+++ ffmpeg/libavcodec/utils.c	(working copy)
@@ -115,7 +115,7 @@
     enum PixelFormat pix_fmt;
 }InternalBuffer;
 
-#define INTERNAL_BUFFER_SIZE 32
+#define INTERNAL_BUFFER_SIZE 64
 
 void avcodec_align_dimensions(AVCodecContext *s, int *width, int *height){
     int w_align= 1;
