Index: ffmpeg/libavcodec/h264.c
===================================================================
--- ffmpeg/libavcodec/h264.c	(revision 14508)
+++ ffmpeg/libavcodec/h264.c	(working copy)
@@ -5335,7 +5335,7 @@
             ctx = coeff_abs_levelgt1_ctx[node_ctx] + abs_level_m1_ctx_base;
             node_ctx = coeff_abs_level_transition[1][node_ctx];
 
-            while( coeff_abs < 15 && get_cabac( CC, ctx ) ) {
+            while( __builtin_expect(coeff_abs < 15 && get_cabac( CC, ctx ), 0) ) {
                 coeff_abs++;
             }
 
