Index: ffmpeg/libavcodec/h264.c
===================================================================
--- ffmpeg/libavcodec/h264.c	(revision 14508)
+++ ffmpeg/libavcodec/h264.c	(working copy)
@@ -1385,7 +1385,7 @@
     }
 
     bufidx = h->nal_unit_type == NAL_DPC ? 1 : 0; // use second escape buffer for inter data
-    h->rbsp_buffer[bufidx]= av_fast_realloc(h->rbsp_buffer[bufidx], &h->rbsp_buffer_size[bufidx], length);
+    h->rbsp_buffer[bufidx]= av_fast_realloc(h->rbsp_buffer[bufidx], &h->rbsp_buffer_size[bufidx], length+FF_INPUT_BUFFER_PADDING_SIZE);
     dst= h->rbsp_buffer[bufidx];
 
     if (dst == NULL){
@@ -1409,6 +1409,8 @@
         dst[di++]= src[si++];
     }
 
+    memset(dst+di, 0, FF_INPUT_BUFFER_PADDING_SIZE);
+
     *dst_length= di;
     *consumed= si + 1;//+1 for the header
 //FIXME store exact number of bits in the getbitcontext (it is needed for decoding)
