Index: configure
===================================================================
--- configure	(revision 5649)
+++ configure	(working copy)
@@ -226,8 +226,10 @@
     log check_ld "$@"
     cat >$TMPC
     log_file $TMPC
-    log $cc $CFLAGS $LDFLAGS "$@" -o $TMPE $TMPC $extralibs
-    $cc $CFLAGS $LDFLAGS "$@" -o $TMPE $TMPC $extralibs >>$logfile 2>&1
+    log $cc $CFLAGS "$@" -c -o $TMPO $TMPC >>$logfile 2>&1
+    $cc $CFLAGS "$@" -c -o $TMPO $TMPC >>$logfile 2>&1
+    log $cc $LDFLAGS "$@" -o $TMPE $TMPO $extralibs >>$logfile 2>&1
+    $cc $LDFLAGS "$@" -o $TMPE $TMPO $extralibs >>$logfile 2>&1
 }
 
 check_cflags(){
